### DISCUSSION TITLE
About addresses in Solana Transactions

### DISCUSSION THREAD

**Discussion by YingJie0712** [trustwallet/wallet-core#2306] [Q&A] [answered]
In solana token transfer, have 3 params: tokenMintAddress/recipientTokenAddress/senderTokenAddress.
How to get them?

**Comment by hewigovens** [comment]
You need to query Solana node, for example https://docs.solana.com/developing/clients/jsonrpc-api#gettokenaccountsbyowner

**Comment by korinsek** [comment, marked as answer]
1. for your wallet you can get solana address out of hdWallet: `hdWallet.getAddress(Blockchain.SOLANA)`. If you sending to recipient I assume he will share his solana address with you.
2. `tokenMintAddress` - you can define your private API where you define currency definitions of your supported SPL tokens (or hardcode it is your preferred way). You can probably get it also direct from https://raw.githubusercontent.com/solana-labs/token-list/main/src/tokens/solana.tokenlist.json.
3. Once you have senders/recipients solana address and tokenMintAddress you can derive recipients/senders tokenAccountAddress with `SolanaAddress(solanaAddress).defaultTokenAddress(tokenMintAddress)`. More details also here: https://spl.solana.com/associated-token-account
